### PR TITLE
Handle a few logs fields with inconsistent data types

### PR DESCRIPTION
**Pull request by prratek**
In similar spirit to #4, the change here modified the `post_process()` method for the logs stream to handle two cases of inconsistent data types:
1. `details.error`, a field of type `object` appears to be an empty string when null so we coerce that to `None` type
2. `response.body` can be either an object or an array of objects depending on the number of elements returned in that fields. When it isn't an array, we convert to an array of length 1 containing the object returned.

### CODE REVIEW

File: tap_auth0/streams.py [hunk 1]
```diff
@@ -169,10 +169,19 @@ def validate_response(self, response: requests.Response) -> None:
...
             row.update({ "scope": scope.split() })
+        details_error = row.get("details", {}).get("error")
+        if isinstance(details_error, str):
+            if details_error == "":
+                row["details"].update({ "error": None })
+            else:
+                row["details"].update({ "error": {"message": details_error} })
+        response_body = row.get("response", {}).get("body")
+        if isinstance(response_body, dict):
+            row["response"].update({ "body": [response_body] })
```
Comment (ReubenFrankel) [inline tap_auth0/streams.py:186]: Looking through our logs and at the [Auth0 logs resource reference](https://auth0.com/docs/api/management/v2#!/Logs/get_logs), I don't see a top-level `response` field. Are you referring to `details.response.body`?

**Comment by ReubenFrankel** [comment]
Hi @prratek, thanks for opening this PR! I've left some comments on your changes. Could you also provide some example log data that demonstrates these inconsistent types for further context?

**Comment by ReubenFrankel** [comment]
[Initial Slack discussion](https://meltano.slack.com/archives/C01PKLU5D1R/p1648495242209059)

**Comment by ReubenFrankel** [comment]
Related Meltano `singer-sdk` issue: https://github.com/meltano/sdk/issues/792

**Comment by ReubenFrankel** [comment]
Only now just realising I never published my review comments, sorry (newer to GitHub back then)! Are you still interested in getting this in? No worries if not.

**Comment by prratek** [comment]
> Only now just realising I never published my review comments, sorry (newer to GitHub back then)! Are you still interested in getting this in? No worries if not.

Thanks for the review! We recently moved away from Auth0 so aren't using the tap anymore. I wouldn't find time to respond to feedback and get this in but feel free to use any of the code in the PR if you'd like.

**Comment by ReubenFrankel** [comment]
Fair enough, I'm gonna close it for now since we don't have a reproducible example of our own, but thanks for coming back regardless. I'll reopen this if anyone else comes along with a similar issue. :slightly_smiling_face:

**Pull request closed without merging**